modules/aws/ec2/main.tf

```diff
@@ -36,30 +36,30 @@ resource "aws_iam_role" "instance_role" {
   EOF
 }
 
-resource "aws_iam_role_policy_attachment" "instance_role" {
+resource "aws_iam_role_policy_attachment" "this" {
   for_each = toset([
     "arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM",
     "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"
   ])
-  role       = aws_iam_role.instance_role.name
+  role       = aws_iam_role.this.name
   policy_arn = each.value
 }
 
-resource "tls_private_key" "private_key" {
+resource "tls_private_key" "this" {
```

Comment:
I would name this on the lines of 'self_generated'
This could be misinterpreted as always being the key which in some scenarios it won't be (e.g. You provide your own key)